From 39138fa1f7185130060f4283bdef11618959922e Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 4 Feb 2025 15:30:14 +0100
Subject: [PATCH 1599/1695] dts: stm32: use PWR through EXTI on stm32mp23x

The hierarchical stack is:
- GIC
- PWR (only on EXTI lines for wakeup pins)
- EXTI
- irq client

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f4329c453e7341a04b60c0cd224f76ad142a3b74 dts: stm32: use PWR through EXTI on stm32mp23x]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Iba7ccda91924bcb6cf235658a1626f680380c10b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434941
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi    | 14 +++++++-------
 core/arch/arm/dts/stm32mp235f-dk.dts |  2 +-
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 94c39b6b7..2c6f44379 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -827,7 +827,6 @@
 		pwr: pwr@44210000 {
 			compatible = "st,stm32mp25-pwr";
 			reg = <0x44210000 0x400>;
-			wakeup-parent = <&exti1 0xffff 0>;
 			interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
@@ -891,6 +890,7 @@
 
 		exti1: interrupt-controller@44220000 {
 			compatible = "st,stm32mp1-exti", "syscon";
+			wakeup-parent = <&pwr>;
 			interrupt-controller;
 			#interrupt-cells = <2>;
 			reg = <0x44220000 0x400>;
@@ -947,12 +947,12 @@
 				<&intc GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
 				<&intc GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_50 */
 				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
+				<&pwr 0 IRQ_TYPE_EDGE_FALLING>,
+				<&pwr 1 IRQ_TYPE_EDGE_FALLING>,
+				<&pwr 2 IRQ_TYPE_EDGE_FALLING>,
+				<&pwr 3 IRQ_TYPE_EDGE_FALLING>,
+				<&pwr 4 IRQ_TYPE_EDGE_FALLING>,
+				<&pwr 5 IRQ_TYPE_EDGE_FALLING>,
 				<0>,
 				<&intc GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>,
 				<0>,						/* EXTI_60 */
diff --git a/core/arch/arm/dts/stm32mp235f-dk.dts b/core/arch/arm/dts/stm32mp235f-dk.dts
index 1bcc40e0d..27877de67 100644
--- a/core/arch/arm/dts/stm32mp235f-dk.dts
+++ b/core/arch/arm/dts/stm32mp235f-dk.dts
@@ -72,7 +72,7 @@
 	pmic2: stpmic2@33 {
 		compatible = "st,stpmic2";
 		reg = <0x33>;
-		interrupts-extended = <&pwr 0 IRQ_TYPE_EDGE_FALLING>;
+		interrupts-extended = <&exti1 52 IRQ_TYPE_EDGE_FALLING>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
 		st,notif-it-id = <0 1>;
-- 
2.47.2

